variables-bastion.tf

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+variable "create_bastion" {
+  default     = true
+  description = "Whether to create a bastion host."
+  type        = bool
+}
+
+variable "bastion_public_ip" {
+  default     = null
+  description = "The IP address of an existing bastion host, if create_bastion = false."
+  type        = string
+}
+
+variable "bastion_allowed_cidrs" {
+  default     = ["0.0.0.0/0"]
+  description = "A list of CIDR blocks to allow SSH access to the bastion host."
+  type        = list(string)
+}
+
+variable "bastion_availability_domain" {
+  default     = null
+  description = "The availability domain for bastion placement. Defaults to first available."
+  type        = string
+}
+
+variable "bastion_nsg_ids" {
+  description = "An additional list of network security group (NSG) IDs for bastion security."
+  default     = []
+  type        = list(string)
+}
+
+variable "bastion_user" {
+  default     = "opc"
+  description = "User for SSH access through bastion host."
+  type        = string
+}
+
+variable "bastion_image_id" {
+  default     = null
+  description = "Image ID for created bastion instance."
+  type        = string
+}
+
+variable "bastion_image_type" {
+  default     = "platform"
+  description = "Whether to use a platform or custom image for the created bastion instance. When custom is set, the bastion_image_id must be specified."
+  type        = string
+  validation {
+    condition     = contains(["custom", "platform"], var.bastion_image_type)
+    error_message = "Accepted values are custom or platform"
+  }
+}
+
+variable "bastion_image_os" {
+  default     = "Oracle Linux"
+  description = "Bastion image operating system name when bastion_image_type = 'platform'."
+  type        = string
+}
+
+variable "bastion_image_os_version" {
+  default     = "8"
```

Comment:
Default should be Autonomous Linux 7.9.

Reply:
I think we should prefer consistency with supported worker images using an OL8 platform image by default and minimal implementation. There will be a good case for some to disable this in favor of the external module for advanced functionality, and this should be easily accomplished by providing its output IP to the optional var here. As with the operator (even more-so), the basic need for most is met by a single lightweight Compute instance, and this relieves the module from passing through a dozen variables not directly related to its purpose + needing to test those input variations. I would've preferred the service if not for that session time limitation...

Reply:
Discussed and in agreement for now - will revisit use of bastion module in a subsequent PR.